### PR TITLE
feat(task-framework): handle executor endpoint for PULL monitor mode

### PR DESCRIPTION
**Pull request by yizhouxw**
## main changes

invovle ExecutorEndpointManager for PULL task monitor mode
- will refresh executor point from K8sJobClient for K8S runMode
- ExecutorHostAdapter is designed for self defined network solution extension
- K8bJobClient add podIdAddress in response for support PULL task monitor mode
- while build K8sJobCaller, set ODC_EXECUTOR_PORT/REPORT_ENABLED env variables
- offer a hard code executor listen port, 8080 by default

invovle JobCredentialProvider for separated metadb and cloud object storage in task executor
- return same credential which is used by odc server by default
- developer may offer other implementation for self define executor metadb and cloud object storage

related enhancement
- change jobProperties from SingleJobProperties to Map<String,String> for easy to extends features.
- move CloudProvider to package service.cloud.model, for reuse outside of cloud object storage part.
 

## other changes
1. update CODEOWNERS
2. add JdbcUrlParser utils, may referenced in other plugins
3. some code refine

### CODE REVIEW

File: server/odc-common/src/main/java/com/oceanbase/odc/common/jdbc/JdbcUrlParser.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import com.oceanbase.odc.common.util.StringUtils;
+
+import lombok.Data;
+import lombok.NonNull;
+
+public class JdbcUrlParser {
```
[the remark this answers is not in the document]
Comment (yizhouxw) [reply server/odc-common/src/main/java/com/oceanbase/odc/common/jdbc/JdbcUrlParser.java]: Rename to SimpleJdbcUrlParser

File: server/plugins/connect-plugin-api/src/main/java/com/oceanbase/odc/plugin/connect/api/JdbcUrlParser.java [hunk 2]
```diff
@@ -13,11 +13,13 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package com.oceanbase.odc.core.shared.jdbc;
+package com.oceanbase.odc.plugin.connect.api;
 
 import java.util.List;
 import java.util.Map;
 
+import com.oceanbase.odc.core.shared.jdbc.HostAddress;
```
Comment (yhilmare) [inline server/plugins/connect-plugin-api/src/main/java/com/oceanbase/odc/plugin/connect/api/JdbcUrlParser.java:21]: HostAddress may be also moved here

File: server/odc-common/src/main/java/com/oceanbase/odc/common/jdbc/JdbcUrlParser.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (yizhouxw) [reply server/odc-common/src/main/java/com/oceanbase/odc/common/jdbc/JdbcUrlParser.java]: change to  SimpleJdbcUrlParser  now

File: server/plugins/connect-plugin-api/src/main/java/com/oceanbase/odc/plugin/connect/api/JdbcUrlParser.java (hunk 2 above)
Comment (yizhouxw) [reply server/plugins/connect-plugin-api/src/main/java/com/oceanbase/odc/plugin/connect/api/JdbcUrlParser.java]: okay, moved now

**Review by yhilmare — APPROVED** [review]
lgtm

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/util/JobUtils.java [hunk 3]
```diff
@@ -132,14 +133,15 @@ public static boolean isProcessRunModeOfEnv() {
     public static ConnectionConfig getMetaDBConnectionConfig() {
         ConnectionConfig config = new ConnectionConfig();
         config.setHost(System.getProperty(JobEnvKeyConstants.ODC_EXECUTOR_DATABASE_HOST));
```
Comment (guowl3) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/task/util/JobUtils.java:135]: environment variable could expire while the main application is still running?

Comment (yizhouxw) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/task/util/JobUtils.java:135]: may occur in process mode scenario, let's trace in separated issue

**Review by guowl3 — APPROVED** [review]
lgtm

**Pull request merged**